core/src/main/java/tc/oc/pgm/enderchest/EnderChestModule.java

```diff
@@ -0,0 +1,74 @@
+package tc.oc.pgm.enderchest;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import java.util.logging.Logger;
+import org.jdom2.Attribute;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.region.Region;
+import tc.oc.pgm.filters.parse.FilterParser;
+import tc.oc.pgm.regions.RegionParser;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class EnderChestModule implements MapModule {
+
+  private final boolean enabled;
+  private final int rows;
+  private final List<Dropoff> dropoffs;
+
+  public EnderChestModule(boolean enabled, int rows, List<Dropoff> dropoffs) {
+    this.enabled = enabled;
+    this.rows = rows;
+    this.dropoffs = dropoffs;
+  }
+
+  @Override
+  public MatchModule createMatchModule(Match match) {
+    return new EnderChestMatchModule(match, enabled, rows, dropoffs);
+  }
+
+  public static class Factory implements MapModuleFactory<EnderChestModule> {
+    @Override
+    public EnderChestModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      FilterParser filters = factory.getFilters();
+      RegionParser regions = factory.getRegions();
+
+      boolean enabled = false;
+      int rows = 3;
+      List<Dropoff> dropoffs = Lists.newArrayList();
+
+      for (Element enderRootEl : doc.getRootElement().getChildren("enderchest")) {
```

Comment:
Use the XMLUtils flattening method here, it will allow support for nested classes etc